### PR TITLE
[Fix]: Allow empty object lists in high-frequency-telemetry YANG model

### PR DESCRIPTION
**Pull request by Pterosaur**
#### Why I did it
We must have a dummy HFT table to support dynamically configuring HFT. This Yang model change is to support the dummy table.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
- Remove mandatory constraint for object_names list
- Add test cases for empty lists and validation scenarios
- Include high-frequency telemetry model
- 
#### How to verify it

Check Azp status. This PR includes its test.

#### Which release branch to backport (provide reason below if selected)

- [x] 202412

#### Tested branch (Please provide the tested image version)

master

### CODE REVIEW

File: src/sonic-yang-models/setup.py [hunk 1]
```diff
@@ -146,6 +146,7 @@
     'sonic-smart-switch.yang',
     'sonic-spanning-tree.yang',
     'sonic-srv6.yang',
+    'sonic-high-frequency-telemetry.yang',
```
Comment (wen587) [inline src/sonic-yang-models/setup.py]: let's put by alphabet order

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/high_frequency_telemetry.json [hunk 2]
```diff
@@ -120,6 +130,110 @@
...
+                    },
+                    {
+                        "name": "Ethernet4",
+                        "lanes": "4",
+                        "speed": 25000
+                    }
+                ]
+            }
+        },
+        "sonic-high-frequency-telemetry:sonic-high-frequency-telemetry": {
```
Comment (wen587) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/high_frequency_telemetry.json:214]: could you add a test for empty object list? I don't think current YANG allowed empty list becasue there are still constraints here if group_name is PORT:
                leaf-list object_names {
                   type string;
                   must "( ../group_name = 'PORT' and current() = /port:sonic-port/port:PORT/port:PORT_LIST/port:name )"
                        + " or ( ../group_name = 'BUFFER_POOL' and current() = /bpl:sonic-buffer-pool/bpl:BUFFER_POOL/bpl:BUFFER_POOL_LIST/bpl:name )"
                        + " or ( ../group_name = 'INGRESS_PRIORITY_GROUP' and substring-before(current(), '|') = /bpg:sonic-buffer-pg/bpg:BUFFER_PG/bpg:BUFFER_PG_LIST/bpg:port and re-match(substring-after(current(), '|'), '[0-9]+') )"
                        + " or ( ../group_name = 'QUEUE' and substring-before(current(), '|') = /bql:sonic-buffer-queue/bql:BUFFER_QUEUE/bql:BUFFER_QUEUE_LIST/bql:port and re-match(substring-after(current(), '|'), '[0-9]+') )";
                }

File: src/sonic-yang-models/yang-models/sonic-high-frequency-telemetry.yang [hunk 3]
```diff
@@ -93,8 +93,6 @@ module sonic-high-frequency-telemetry {
                         + " or ( ../group_name = 'QUEUE' and substring-before(current(), '|') = /bql:sonic-buffer-queue/bql:BUFFER_QUEUE/bql:BUFFER_QUEUE_LIST/bql:port and re-match(substring-after(current(), '|'), '[0-9]+') )";
                 }
 
-                must "count(object_names) > 0";
-
                 leaf-list object_counters {
```
Comment (wen587) [inline src/sonic-yang-models/yang-models/sonic-high-frequency-telemetry.yang:96]: please also include such config in sample_config_db.json

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/high_frequency_telemetry.json (hunk 2 above)
Comment (Pterosaur) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/high_frequency_telemetry.json:214]: https://github.com/sonic-net/sonic-buildimage/pull/23917/files#diff-e4744a17a9c8acd94344ef9dfdec05bee6b87ce5506dd69e790fba84e7b46eb5R145-R148

This test is for empty object list.

File: src/sonic-yang-models/yang-models/sonic-high-frequency-telemetry.yang (hunk 3 above)
Comment (Pterosaur) [reply src/sonic-yang-models/yang-models/sonic-high-frequency-telemetry.yang:96]: Done

File: src/sonic-yang-models/setup.py (hunk 1 above)
Comment (Pterosaur) [reply src/sonic-yang-models/setup.py]: Done

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/high_frequency_telemetry.json (hunk 2 above)
Comment (wen587) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/high_frequency_telemetry.json:214]: The test in the link still have elements in object_names
```
                 "profile_name": "partial_objects",
                        "group_name": "PORT",
                        "object_names": [
                            "Ethernet0"
                        ],
```
I mean empty object_names what ndm will generates. such as:
```
   "profile_name": "partial_objects",
                        "group_name": "PORT",
                        "object_names": [],
```
or
```
 "profile_name": "partial_objects",
                        "group_name": "PORT",
                        "object_names": "",
```

Comment (Pterosaur) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/high_frequency_telemetry.json:214]: It seems that our SONiC YANG parser doesn't support the structure like:
```
"object_names": [],
```
It will raise an error from Yang parser.
```
libyang[0]: Invalid JSON data (unexpected value). (path: /sonic-high-frequency-telemetry:sonic-high-frequency-telemetry/HIGH_FREQUENCY_TELEMETRY_GROUP/HIGH_FREQUENCY_TELEMETRY_GROUP_LIST[profile_name='dummy_profile'][group_name='PORT']/object_names)
 Exception >Invalid JSON data (unexpected value). in not empty< in /sonic/src/sonic-yang-models/tests/yang_model_tests/test_yang_model.py:278
```
IMHO, its schema is defined by us, so we can define we only support a json without these fields.

Comment (wen587) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/high_frequency_telemetry.json:214]: Hi @Pterosaur , right. That's what i am concerned. e.g. if NDM liquid provides "" for object_names and object_counters. Such config will still fail yang validation after we remove `must "count(object_names) > 0";` . 

liquid is using "" for field value:
https://msazure.visualstudio.com/One/_git/Networking-Metadata?path=/src/data/Network/DeviceConfigTemplates/Sonic_HIGH_FREQUENCY_TELEMETRY_GROUP_Template.liquid&version=GBmaster&_a=contents

To support such field for NDM, should we remove the constraints? such as 
```
                leaf-list object_names {
                   type string;
                   must "( ../group_name = 'PORT' and current() = /port:sonic-port/port:PORT/port:PORT_LIST/port:name )"
```
 I am not sure if "" applies to leaf-list. or we should pick [] instead for NDM

Comment (Pterosaur) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/high_frequency_telemetry.json:214]: Gotch, I add two negative test cases, please check it. We need to update the NDM code later.

File: src/sonic-yang-models/yang-models/sonic-high-frequency-telemetry.yang [hunk 4]
```diff
@@ -93,8 +93,6 @@ module sonic-high-frequency-telemetry {
                         + " or ( ../group_name = 'QUEUE' and substring-before(current(), '|') = /bql:sonic-buffer-queue/bql:BUFFER_QUEUE/bql:BUFFER_QUEUE_LIST/bql:port and re-match(substring-after(current(), '|'), '[0-9]+') )";
                 }
 
-                must "count(object_names) > 0";
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-high-frequency-telemetry.yang:96]: https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-pbh.yang#L119

Can we use min-elements?

Comment (Pterosaur) [reply src/sonic-yang-models/yang-models/sonic-high-frequency-telemetry.yang:96]: Use min-elements for what? I removed this condition because I want to support no explicate fields in HFT_GROUP object. such as:
```
"dummy_table|PORT": {}
```

**Review by ganglyu — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]
